dynamiqs/integrators/apis/mcsolve.py

```diff
@@ -0,0 +1,272 @@
+from __future__ import annotations
+
+import logging
+from functools import partial
+
+import jax
+import jax.numpy as jnp
+import optimistix as optx
+from jax import Array
+from jaxtyping import ArrayLike
+from optimistix import AbstractRootFinder
+
+from ..._checks import check_shape, check_times
+from ...gradient import Gradient
+from ...options import Options
+from ...qarrays.qarray import QArray, QArrayLike
+from ...qarrays.utils import asqarray
+from ...result import MCSolveResult
+from ...solver import Dopri5, Dopri8, Euler, Kvaerno3, Kvaerno5, Solver, Tsit5
+from ...time_array import TimeArray
+from .._utils import (
+    _astimearray,
+    cartesian_vmap,
+    catch_xla_runtime_error,
+    get_integrator_class,
+    multi_vmap,
+)
+from ..mcsolve.diffrax_integrator import (
+    MCSolveDopri5Integrator,
+    MCSolveDopri8Integrator,
+    MCSolveEulerIntegrator,
+    MCSolveIntegrator,
+    MCSolveKvaerno3Integrator,
+    MCSolveKvaerno5Integrator,
+    MCSolveTsit5Integrator,
+)
+
+
+def mcsolve(
+    H: QArrayLike | TimeArray,
+    jump_ops: list[QArrayLike | TimeArray],
+    psi0: QArrayLike,
+    tsave: ArrayLike,
+    *,
+    keys: ArrayLike = jax.random.split(jax.random.key(31), num=10),  # noqa: B008
+    exp_ops: list[QArrayLike] | None = None,
+    solver: Solver = Tsit5(),  # noqa: B008
+    root_finder: AbstractRootFinder | None = optx.Newton(1e-5, 1e-5, optx.rms_norm),  # noqa: B008
+    gradient: Gradient | None = None,
+    options: Options = Options(),  # noqa: B008
+) -> MCSolveResult:
+    r"""Perform Monte-Carlo evolution, unraveling the master equation.
```

Comment:
```suggestion
    r"""Solve the Lindblad master equation through Monte-Carlo sampling of its jump unraveling.
```
(maybe the formulation is not 100% exact and could be improved, the comment is mostly to use the "Solve ..." similar to other functions)